### PR TITLE
Prove kernelc tree examples

### PR DESCRIPTION
**Pull request by yilongli**
more to come...

### CODE REVIEW

File: include/z3/prelude.smt2 [hunk 1]
```diff
@@ -17,10 +17,12 @@
 (define-fun smt_set_lt  ((s1 IntSet) (s2 IntSet)) Bool (forall ((i Int) (j Int)) (implies (>= i j) (not (and (select s1 i) (select s2 j))))))
 (define-fun smt_set_le  ((s1 IntSet) (s2 IntSet)) Bool (forall ((i Int) (j Int)) (implies (>  i j) (not (and (select s1 i) (select s2 j))))))
 
+; bool to int
+(declare-fun smt_bool2int ((Bool)) Int)
```
Comment (andreistefanescu) [inline include/z3/prelude.smt2]: Replace it with `(define-fun smt_bool2int ((b Bool)) Int (ite b 1 0))` and you won't be needing the lemmas anymore.

File: samples/kernelc/patterns/tree_pattern.k [hunk 2]
```diff
@@ -64,5 +68,12 @@ module TREE-PATTERN
     => tree_keys(TL) <IntSet { I } andBool { I } <IntSet tree_keys(TR)
        andBool bst(TL) andBool bst(TR)
   rule bst(leaf) => true
-endmodule
 
+  // TODO(YilongL): should probably go somewhere else; needs a better name...
+  syntax Int ::= bool2Int(Bool)                 [function, smtlib(smt_bool2int)]
+  rule bool2Int(true) => 1
+  rule bool2Int(false) => 0
+  rule bool2Int(B) =/=K 0 => bool2Int(B) ==K 1          [lemma]
```
Comment (andreistefanescu) [inline samples/kernelc/patterns/tree_pattern.k]: see above

**Comment by andreistefanescu** [comment]
lgtm

**Comment by yilongli** [comment]
Jenkins: retest this please

**Comment by yilongli** [comment]
Since I cannot prove more tree examples without reimplementing `BuiltinList`, I would like to merge this pull request after all tests pass.

**Comment by andreistefanescu** [comment]
lgtm

**Pull request merged**